functionary/prompt_template/base_template.py

```diff
@@ -11,6 +12,128 @@
 class PromptTemplate:
     _instance = None
 
+    @abstractmethod
+    def get_start_of_function_call_token(self) -> str:
+        """returns a token that indicates the start of a function call in the prompt template
+        Returns:
+            str: a string token
+        """
+        raise NotImplementedError
+
+    @abstractmethod
+    def get_stopping_token(self, stage: Literal["function", "parameter"]) -> int:
```

Comment:
Can we rename get_stopping_token  to avoid confusion with: def get_stop_tokens_for_generation?

Reply:
Gotcha